test/Tests/src/Semantic/Any_Spec.enso

```diff
@@ -42,19 +42,64 @@ spec =
             My_Type.x . should_equal "Any:My_Type"
             With_X.x . to_text . should_equal "Any:With_X"
-            With_X.y . to_text . should_equal "With_X.type.y[Any_Types.enso:12-32]"
+            Meta.is_a With_X.y Function . should_be_true
             With_Y.x . to_text . should_equal "Any:With_Y"
-            With_Y.y . to_text . should_equal "With_Y.type.y[Any_Types.enso:18-38]"
+            Meta.is_a With_Y.y Function . should_be_true
             With_X.to_text . to_text . should_equal "With_X"
             With_Y.to_text . to_text . should_equal "With_Y"
+            Any.x . to_text . should_equal "Any:Any"
+            Any.x self=Any . to_text . should_equal "Any:Any"
             Any.x self=With_X . should_equal "Any:With_X"
             Any.x self=With_Y . should_equal "Any:With_Y"
             Any.x (My_Type.Value 22) . should_equal "Any:(My_Type.Value 22)"
             Any.x (With_X.Value 22) . should_equal "Any:(With_X.Value 22)"
             Any.x (With_Y.Value 22) . should_equal "Any:With_Y(22)"
             Date.to_display_text . should_equal "Date"
 
+        Test.specify "static method calls on Any should have defaulted self argument to Any" <|
+            # Any.== is a method that takes two arguments (self other)
+            Meta.is_a (Any.==) Function . should_be_true
+            Meta.is_a (Any.== self=Boolean) Function . should_be_true
+            (Any.== self=Any Boolean) . should_be_false
+            (Any.== Any Boolean) . should_be_false
+            (Any.== Boolean) . should_be_false
+            (Any.== Any) . should_be_true
+            (Any == Boolean) . should_be_false
+            (Any == Any) . should_be_true
+
+            # Any.to_text is a method that takes one argument (self)
+            Any.to_text . should_equal "Any"
+            Any.to_text self=Any . should_equal "Any"
+            Any.to_text self=Vector . should_equal "Vector"
+            Any.to_text self=[1,2,3] . should_equal "[1, 2, 3]"
+            Any.to_text Any . should_equal "Any"
+            Any.to_text Vector . should_equal "Vector"
+            Any.to_text [1,2,3] . should_equal "[1, 2, 3]"
+
+            # Any.spec_method is a method that takes three arguments (self, arg1, arg2)
+            Meta.is_a (Any.spec_method) Function . should_be_true
+            Meta.is_a (Any.spec_method Boolean) Function . should_be_true
+            Meta.is_a (Any.spec_method self=Boolean) Function . should_be_true
+            Meta.is_a (Any.spec_method self=Boolean Vector) Function . should_be_true
+            Any.spec_method Boolean Vector . should_equal "Any.spec_method:{Any}{Boolean}{Vector}"
+            Any.spec_method Any Boolean Vector . should_equal "Any.spec_method:{Any}{Boolean}{Vector}"
+            Any.spec_method Date Boolean Vector . should_equal "Any.spec_method:{Date}{Boolean}{Vector}"
+            Any.spec_method self=Any Boolean Vector . should_equal "Any.spec_method:{Any}{Boolean}{Vector}"
```

Comment:
This behaviour looks very weird, but I guess it is the best we can do to satisfy the requirements we want from the static methods.

Reply:
I'd add:
```
Any.spec_method self=Date Boolean Vector . should_equal "Any.spec_method:{Date}{Boolean}{Vector}"
```
that's my preferred way to invoke `Any` method with explicit `self`.

Reply:
Added in d4f0b6d239a750aba1aa3291145b315e7ee47851